File: api/Models/Hotel.cs
﻿using System.Collections.Generic;

namespace iTechArt.Hotels.Api
{
    public class Hotel
    {
        public Hotel()
        {
            Images = new HashSet<Image>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string Country { get; set; }
        public string City { get; set; }
        public string Address { get; set; }
        public string Description { get; set; }
        public int? MainImageId { get; set; }

        public ICollection<Image> Images { get; set; }
    }
}

File: admin/src/app/Dtos/hotel.ts
import { Facility } from './facility';

export interface Hotel {
  id: number;
  name: string;
  country: string;
  city: string;
  address: string;
  description?: string;
  mainImageId?: number;
  facilities: Facility[];
  checkInTime: string;
  checkOutTime: string;
}

export type HotelToEdit = Omit<Hotel, 'id'>;

export type HotelToAdd = Omit<Hotel, 'id' | 'mainImageId' | 'facilities'>;
